File: go_chunk_app/simpleversion.go
package main

import (
	"strings"
)

func threeWordsChunk(sentence string) string {
	words := strings.Fields(sentence)
	var chunks []string

	for i := 0; i <= len(words)-3; i++ {
		chunks = append(chunks, strings.Join(words[i:i+3], " "))
	}

	return strings.Join(chunks, " | ")
}

File: rust_chunk_app/src/simple_version.rs
// This file contains the simple version of the chunking logic

pub fn three_words_chunk(sentence: &str) -> String {
    let words: Vec<&str> = sentence.split_whitespace().collect();
    let chunks: Vec<String> = words.windows(3)
        .map(|chunk| chunk.join(" "))
        .collect();
    chunks.join(" | ")
}